CRM/PivotCache/GroupActivity.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/**
+ * Manages 'pivotreport' cache group.
+ */
+class CRM_PivotCache_GroupActivity extends CRM_PivotCache_AbstractGroup {
+
+  public function __construct($name = NULL) {
+    $this->name = 'pivotreport.Activity';
+  }
+
+  /**
+   * Gets DAO resource of cached data for specified criteria.
+   *
+   * @param int $page
+   * @param array $params
+   *
+   * @return \CRM_Core_DAO
+   */
+  public function query($page, array $params) {
+    $cache = new CRM_Core_DAO_Cache();
+
+    $cache->group_name = $this->getName();
+
+    if (!empty($params['start_date'])) {
+      $whereStartDate = CRM_Core_DAO::createSQLFilter(
+        'path',
+        array(
+          '>=' => 'data_' . substr($params['start_date'], 0, 10) . '_' . str_pad($page, 6, '0', STR_PAD_LEFT),
```

Comment:
It seems this is the same logic from `AbstractGroup::getPath`, isn't it? What about making that method protected, so that it can be used by child classes, and use it in here?